examples/rcn.py

```diff
@@ -0,0 +1,392 @@
+from scipy.ndimage import maximum_filter
+from scipy.signal import fftconvolve
+from sklearn.datasets import fetch_openml
+
+from pgmax.fg import graph, groups
+
+os.environ["XLA_PYTHON_CLIENT_ALLOCATOR"] = "platform"
+os.environ["XLA_PYTHON_CLIENT_PREALLOCATE"] = "false"
+
+# %% [markdown]
+# # 1. Load the data
+# %%
+hps, vps = 12, 12
+train_size = 20
+test_size = 20
+
+
+def fetch_mnist_dataset(train_size: int, test_size: int, seed: int = 5):
+    """Returns training and test images sampled randomly from the set of MNIST images.
+    Args:
+        train_size: Desired number of training images.
+        test_size: Desired number of test images.
+    Returns:
+        train_set: A list of length train_size containing images from the MNIST train dataset.
+        train_labels: Corresponding labels for the train images.
+        test_set: A list of length test_size containing images from the MNIST test dataset.
+        test_labels: Corresponding labels for the test images.
+    """
+
+    mnist_train_size = 60000
+    dataset = fetch_openml("mnist_784", as_frame=False, cache=True)
+
+    print("Fetched the data")
+    mnist_images = dataset["data"]
+    mnist_labels = dataset["target"].astype("int")
+
+    def _sample_data(images, labels, num_per_class):
+        t_set = []
+        t_labels = []
+        for i in range(10):
+            idxs = np.random.choice(np.argwhere(labels == i)[:, 0], num_per_class)
+            for idx in idxs:
+                img = images[idx].reshape(28, 28)
+                img_arr = jax.image.resize(
+                    image=img, shape=(112, 112), method="bicubic"
+                )
+                img = jnp.pad(
+                    img_arr,
+                    pad_width=tuple([(p, p) for p in (44, 44)]),
+                    mode="constant",
+                    constant_values=0,
+                )
+
+                t_set.append(img)
+                t_labels.append(i)
+        return t_set, t_labels
+
+    np.random.seed(seed)
+    full_train_set, full_train_labels = (
+        mnist_images[:mnist_train_size],
+        mnist_labels[:mnist_train_size],
+    )
+    full_test_set, full_test_labels = (
+        mnist_images[mnist_train_size:],
+        mnist_labels[mnist_train_size:],
+    )
+
+    train_set, train_labels = _sample_data(
+        full_train_set, full_train_labels, train_size // 10
+    )
+    test_set, test_labels = _sample_data(
+        full_test_set, full_test_labels, test_size // 10
+    )
+
+    return train_set, np.array(train_labels), test_set, np.array(test_labels)
+
+
+# %%
+train_set, train_labels, test_set, test_labels = fetch_mnist_dataset(
+    train_size, test_size
+)
+
+# %% [markdown]
+# # 2. Load the model
+
+# %%
+data = np.load("example_data/rcn.npz", allow_pickle=True, encoding="latin1")
+frcs, edges, suppression_masks, filters = (
+    data["frcs"],
+    data["edges"],
+    data["suppression_masks"],
+    data["filters"],
+)
+
+M = (2 * hps + 1) * (2 * vps + 1) + 1
```

Comment:
Can we get rid of this `+ 1` as previously discussed? The 0 state does not make sense in this context and does not match the `science_rcn` implementation.

Reply:
Changed.